extensions/src/main/java/org/pentaho/platform/web/http/api/resources/ActionResource.java

```diff
@@ -88,14 +86,14 @@ public Response invokeAction(
     @QueryParam( ActionUtil.INVOKER_ACTIONID ) String actionId,
     @QueryParam( ActionUtil.INVOKER_ACTIONCLASS ) String actionClass,
     @QueryParam( ActionUtil.INVOKER_ACTIONUSER ) String actionUser,
-    final String actionParams ) {
+    final ActionParams actionParams ) {
```

Comment:
> I am not sure SystemListener should be invoking ResourceAction explicitly like that. I think it should call to an interface and not the rest endpoint directly.

Absolutely correct. 
In tine ( cleanup process ), we will step away from calling for `new ActionResource().invokeAction( String , .. )`, and into programming to interfaces, via IActionInvoker invoker = PentahoSystem.get( IActionInvoker.class ); invoker.invokeAction( String, ... )